yarn-project/prover-node/src/prover-node.ts

```diff
@@ -250,13 +254,25 @@ export class ProverNode implements ClaimsMonitorHandler, EpochMonitorHandler, Pr
     // Create a processor using the forked world state
     const publicProcessorFactory = new PublicProcessorFactory(this.contractDataSource, this.telemetryClient);
 
+    const epochHash = sha256(Buffer.concat(blocks.map(block => block.hash().toBuffer())));
+    const proverCache = await this.proverCacheManager.openCache(epochNumber, epochHash);
+
     const cleanUp = async () => {
       await publicDb.close();
       await proverDb.close();
+      await this.proverCacheManager.removedStaleCaches(epochNumber);
```

Comment:
Shouldn't we be cleaning up the cache for _this_ epoch here? I understand this call will clean up all caches up to but excluding the `epochNumber` one.

Reply:
I'll check when cleanup gets called. We could, yeah!